src/event_form/steps/event-form-step3.controller.js

```diff
@@ -226,87 +225,68 @@ function EventFormStep3Controller(
         locations[key] = jsonLDLangFilter(locations[key], language, true);
       });
-      $scope.locationsForCity = locations;
-      return locations;
+      // sort locations by Levenshtein distance
+      var sortedLocations = locations.sort(orderSearchedLocations(filterValue));
+      $scope.locationsForCity = sortedLocations;
+      return sortedLocations;
     }
 
-    function clearLoadingState() {
-      $scope.locationsSearched = false;
-      $scope.loadingPlaces = false;
+    function showErrorAndReturnEmptyList () {
+      $scope.locationAutoCompleteError = true;
+      return [];
     }
 
-    $scope.loadingPlaces = true;
-    $scope.locationAutoCompleteError = false;
-
     if ($scope.selectedCountry.code === 'BE') {
       return cityAutocomplete
-        .getPlacesByZipcode(city.zip, 'BE')
-        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList)
-        .finally(clearLoadingState);
+        .getPlacesByZipcode($scope.selectedCityObj.zip, 'BE', filterValue)
+        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList);
     }
+
     if ($scope.selectedCountry.code === 'NL') {
       return cityAutocomplete
-        .getPlacesByCity(city.label, 'NL')
-        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList)
-        .finally(clearLoadingState);
+        .getPlacesByCity($scope.selectedCityObj.name, 'NL')
+        .then(updateLocationsAndReturnList, showErrorAndReturnEmptyList);
     }
+
   };
 
+  $scope.getPlaces = controller.getPlaces;
+
   controller.cityHasLocations = function () {
     return $scope.locationsForCity instanceof Array && $scope.locationsForCity.length > 0;
   };
-  $scope.cityHasLocations = controller.cityHasLocations;
-
-  controller.locationSearched = function () {
-    $scope.locationsSearched = true;
-  };
-  $scope.locationSearched = controller.locationSearched;
 
-  controller.filterCityLocations = function (filterValue) {
-    return function (location) {
-      var words = filterValue.match(/\w+/g).filter(function (word) {
-        return word.length > 2;
-      });
-      var addressMatches = words.filter(function (word) {
-        return location.address.streetAddress.toLowerCase().indexOf(word.toLowerCase()) !== -1;
-      });
-      var nameMatches = words.filter(function (word) {
-        return location.name.toLowerCase().indexOf(word.toLowerCase()) !== -1;
-      });
-
-      return addressMatches.length + nameMatches.length >= words.length;
-    };
-  };
-  $scope.filterCityLocations = controller.filterCityLocations;
+  /**
+   * Order locations by Levenshtein distance
+   *
+   * @param {string} filterValue
+   */
 
-  controller.orderCityLocations = function (filterValue) {
+  function orderSearchedLocations(filterValue) {
     return function (location) {
-      return new Levenshtein(location, location.name + '' + location.address.streetAddress);
+      return new Levenshtein(filterValue, location.name + '' + location.address.streetAddress);
```

Comment:
I'm confused how this could have worked originally if `location` was an object with `name`, `address`, etc and now we're passing a string? Your change does seem to make more sense though since you need to the filter value to calculate the distance... Was this a bug? 😕

Reply:
Not really sure if it worked before... 
I also  think it's strange to concatenate both location name and location.address.streetAddress.